### PR TITLE
`.tsx` files suppot

### PR DESCRIPTION
**Pull request by WojciechMatuszewski**
**What**:
I've added support for `.tsx` files.

**Why**:
First of all, thank you for open-sourcing this package. I wanted to use this package for my own internal-workshop, but I need `.tsx` support, so I decided to open this PR.

**How**:

Only 2 changes required - adding the file extension to `load-files.js` so it's picked up as a file and modifying the condition within `index.js` so that the file is loaded as a component.

**Checklist**:

- [x] Documentation N/A
- [x] Tests
- [x] Ready to be merged

### CODE REVIEW

**Comment by WojciechMatuszewski** [comment]
Done

File: src/index.js [hunk 1]
```diff
@@ -19,8 +19,10 @@ const originalDocumentElement = document.documentElement
 function makeKCDWorkshopApp({imports, filesInfo, projectTitle, options = {}}) {
   const lazyComponents = {}
 
+  const componentExtensions = ['.js', '.md', '.mdx', '.tsx']
```
Comment (kentcdodds) [inline src/index.js:22]: I think we'll want `.ts` here as well

Comment (WojciechMatuszewski) [reply src/index.js:22]: How come? Can `.ts` can be used as an component?

Comment (WojciechMatuszewski) [reply src/index.js:22]: I mean these extensions are either run through loaders to make them renderable or contain JSX. I never seen JSX used within `.ts` (can it even be used inside `.ts`?)

Comment (kentcdodds) [reply src/index.js:22]: You can create a component without tsx is you don't use JSX in the file. I just want to be complete and not surprising for folks.

Comment (WojciechMatuszewski) [reply src/index.js:22]: With the `React.createElement` API right? Makes sense, changing.

**Comment by kentcdodds** [comment]
@all-contributors please add @WojciechMatuszewski for code and tests

**Pull request merged**